src/KM_Controls.pas

```diff
@@ -3447,20 +3493,36 @@ procedure TKMNumericEdit.SetValue(aValue: Integer);
 procedure TKMNumericEdit.ValidateText;
 var
   I: Integer;
+  AllowedChars: TSetOfAnsiChar;
+  OnlyMinus, IsEmpty: Boolean;
 begin
+  IsEmpty := (fText = #8); // When deleting text with Backspace last character is still in string - backspace character (#8)
+
+  AllowedChars := ['0'..'9'];
   //Validate contents
   for I := Length(fText) downto 1 do
-  if not KromUtils.CharInSet(fText[I], ['0'..'9']) then
   begin
-    Delete(fText, I, 1);
-    if CursorPos >= I then //Keep cursor in place
-      CursorPos := CursorPos - 1;
+    if I = 1 then Include(AllowedChars, '-');
+
+    if not KromUtils.CharInSet(fText[I], AllowedChars) then
+    begin
+      Delete(fText, I, 1);
+      if CursorPos >= I then //Keep cursor in place
+        CursorPos := CursorPos - 1;
+    end;
   end;
 
-  if fText = '' then
+  OnlyMinus := (fText = '-');
+
+  if (fText = '') or OnlyMinus or IsEmpty then
     Value := 0
   else
-    Value := StrToInt(fText);
+    SetValueNCheckRange(StrToInt64(fText));
+
+  if OnlyMinus then fText := '-'; //Set text back to '-' while still editing.
+  if IsEmpty then fText := ''; //Set text back to '' while still editing.
+
+  CursorPos := Min(CursorPos, Length(fText)); //In case we had leading zeros in fText string

```

Comment:
fixed small issue:
When we had leading zeros in fText string, cursor jumped to unpropriate position